tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1566,20 +1566,48 @@ Status BinaryTensorOpWeight(OpConverterParams* params,
   return tensorflow::Status::OK();
 }
 
-enum class ConvolutionType { DEFAULT, DEPTHWISE_CONV };
-
-tensorflow::Status ConvertConv2DHelper(OpConverterParams* params, int group) {
+tensorflow::Status ConvertConv2DHelper(OpConverterParams* params, int group,
+                                       bool is_conv2d_backprop_input) {
   const auto& inputs = params->inputs;
   const auto& node_def = params->node_def;
-  if (inputs.size() != 2) {
-    return tensorflow::errors::InvalidArgument("Two inputs are expected for ",
-                                               node_def.op(), ", at ",
-                                               node_def.name());
-  }
-  if (inputs.at(0).is_weights()) {
-    return tensorflow::errors::Unimplemented(
-        node_def.op(), " is only implemented for tensors, not weights, at ",
-        node_def.name());
+  TRT_TensorOrWeights input, backprop_output_size;
+  const nvinfer1::ITensor* tensor = nullptr;
+  // TODO(tmorris): Add helper functions for validating inputs to avoid code
+  // duplication.
+  if (is_conv2d_backprop_input) {
+    // Conv2DBackpropInput has 3 inputs: input_sizes, filter, and out_backprop.
+    // In the case of conv2d_transpose, these correspond to: output size,
```

Comment:
s/conv2d_transpose/conv2d_backprop_input

Reply:
This is actually supposed to be conv2d_transpose, it is explaining how the meanings of the inputs to Conv2DBackpropInput change for conv2d_tranpose

Reply:
sorry, my bad. Did not read the first line.